### ISSUE TITLE
Where to store transparent proxy iptables script?

### ISSUE THREAD

**Issue by thomask77** [yrutschle/sslh#50]
Hi!

I'm using sslh on Debian 8.0 (Jessie) and use the transparent proxy feature.

Where should I put the iptables script to make it persistent? I think there should be a hint added to the (otherwise great) documentation.

**Comment by yrutschle** [comment]
I think this is distribution-dependent. In Debian there is a package called 'iptables-persistent' that I am reasonably sure does what you want :-)

**Comment by thomask77** [comment]
`iptables-persistent` does not seem to save the `ip rule` / `ip route` commands.. also it would save all of the (possibly) other non-sslh rules, too.

In the mean time, I came up with this solution (tested on debian, but should work on other systemd-based distros, too):

  * Create `/usr/local/sbin/sslh-iptables`:
```bash
#!/bin/sh

# Rules for transparent sslh proxy
# https://github.com/yrutschle/sslh
#
iptables -t mangle -N SSLH
iptables -t mangle -A SSLH --jump MARK --set-mark 0x1
iptables -t mangle -A SSLH --jump ACCEPT

iptables -t mangle -A OUTPUT --protocol tcp --out-interface eth0 --sport 4022 --jump SSLH
iptables -t mangle -A OUTPUT --protocol tcp --out-interface eth0 --sport 4443 --jump SSLH

ip rule add fwmark 0x1 lookup 100
ip route add local 0.0.0.0/0 dev lo table 100
```

  * Create `/etc/systemd/system/sslh-iptables.service`:
```ini
[Unit]
Description=iptables script for sslhd transparent proxy

[Service]
ExecStart=/usr/local/sbin/sslh-iptables

[Install]
WantedBy=multi-user.target
```

  * Activate:
```bash
chmod +x /usr/local/sbin/sslh-iptables
systemctl enable sslh-iptables.service
systemctl start sslh-iptables.service
```

Perhaps this could be added to the documentation.
Best regards, thomas

**Comment by thomask77** [comment]
After reading http://www.freedesktop.org/software/systemd/man/systemd.service.html

I think, 

```ini
[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/local/sbin/sslh-iptables
```

.. should be more correct, because this way, the rules are guaranteed to be loaded only once.
A stop-script is still missing.. the rules will be loaded twice (and thus fail) on stop/start or restart.

**Comment by ViKingIX** [comment]
I think that the clean way is to place the scripts in interfaces file. Using the post-up option will group things prettier.

**Comment by yrutschle** [comment]
I assume this is no longer relevant (and still distribution-dependant, rather than sslh-dependant)
